### ISSUE TITLE
Images with transparencies show the placeholder through the transparent bits

### ISSUE THREAD

**Issue by jimjohnsonrollings** [bbc/simorgh#6429]
**Is your feature request related to a problem? Please describe.**
If a journalist adds a PNG with transparency in it to a CPS STY, then Simorgh displays the image with the BBC blocks image placeholder visible through it (on both Canonical and AMP).

<img width="709" alt="Screen Shot 2020-05-06 at 10 53 46" src="https://user-images.githubusercontent.com/8124899/81183154-b4ddcf80-8fa6-11ea-8546-24f57a567425.png">

[Asset where the image above lives.](https://www.bbc.co.uk/news/uk-politics-49560557)

**Describe the solution you'd like**
If the image has a transparency, then the placeholder should not be visible, and the background should match the background of the page.

I suspect there will need to be some accessibility considerations around ensuring the background colour is correct in various reading modes.

**Describe alternatives you've considered**
N/A

**Testing notes**
- Create unit tests to check if placeholder present when a image is passed in.
- Alt test matches the image, rather than placeholder.
- Check colour matches the page background.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**

**Comment by FK78** [comment]
Investigate Lazy Loading, possibly contact @12

**Comment by ghost** [comment]
## The Cause
In simorgh the `<Image>` is wrapped in a `<ImagePlaceholder>` component. `<imagePlaceholder>` provides the BBC placeholder as a background image.
When the image loads, the BBC placehodler is not replaced.

```jsx
<ImagePlaceholder>
  <LazyLoad>
    <Image/>
  </LazyLoad>
</ImagePlaceholder>
```

## Posible Solutions
### 1. Passing `placeholder` prop to `react-lazyload`
[react-lazyload](https://github.com/twobin/react-lazyload) provides a `placeholder` prop that can be used to provide an element to be displayed before a component is rendered. Instead of wrapping `<LazyLoad>` in a `<ImagePlaceholder>`. We could instead pass `<ImagePlaceholder>` as a `placeholder` prop.

```jsx
<LazyLoad placeholder={<ImagePlaceholder />}>
  <Image />
</LazyLoad>
```
### Issues
This cause the shifting of elements when the images load which does not provide good user experience. This is caused by the fact that `react-lazyload` does not wait untill the image is loaded to do the swap.

![ezgif com-resize](https://user-images.githubusercontent.com/48688870/81546669-6142fb80-9383-11ea-9fee-8532733842af.gif)

This could be fixed by wrapping `<Image>` into a styled component that fixes the shifting of elements when an image loads.

```jsx
const ImageWrapper = styled.div`
  display: block;
  position: relative;
  height: 0;
  overflow: hidden;
  padding-bottom: ${props =>
    props.ratio}%; /* https://www.voorhoede.nl/en/blog/say-no-to-image-reflow/ */
`;
```

However this also introduces another problem. The sudden flash of white background when the place holder is swapped out.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/48688870/81546879-af57ff00-9383-11ea-9eb2-24139e8b4a40.gif)

### 2. Adding an `onLoad` listener to `<Image>`
To have the placeholder swapped out with the loaded image without introducing user experience issues is the desired outcome.
We could add an `onLoad` listener to `<Image>` which will be called once the image has finished loading. We could use this to trigger a change to remove `background-image` from `<ImagePlacehoder>`.

```jsx
<ImagePlaceholder>
  <LazyLoad>
    <Image onLoad={()=> changeBackgroundImage()}/>
  </LazyLoad>
</ImagePlaceholder>
```

### 3. Look into lazy loading libraries that are specific to images.

`react-lazyload` is a generic solution for lazy loading. It doesn't track image loading. It will swap out the `placeholder` component immediately the lazyload threshold is met.

**Comment by chris-hinds** [comment]
Thanks for the detailed writeup @sadickisaac 

Are you able to do a small investigation into a couple of lazy-loading libraries that are specific to images they might solve this issue without impeding the user experience?

I think its best to ensure we are using the correct library for the job before we try an implement our own fixes on-top of the existing library

**Comment by 12** [comment]
Thanks for the write-up @sadickisaac. 

Though to me, this looks like it can be solved simply by adding a white background to images loaded inside the lazyloaded component... 

~#6493~

bbc/psammead#3469

**Comment by jimjohnsonrollings** [comment]
* Not white.  The background of the page is subtly off-white.  Worth noting that UX are also thinking about Dark Mode - so we may need to return to this in the future and make the background the relevant dark colour, but that's not one for today.

**Comment by 12** [comment]
Thanks @jimjohnsonrollings, I have noted that in the POC I mocked up.

What I think we could and probably should do in that case:

1. Update `@bbc/psammead-image` to accept a `colour` prop, to denote what background colour the image should have. If `null`, no background is needed.
2. Update `@bbc/psammead-image` to remove **any** background colour from CSS if the image fails to load. This can be done with an `onError` handler. This is so that the image `alt` text still appears inside the placeholder container, with no different background attached.
3. Pull this into Simorgh. 
4. Pass a `colour` prop to the new version of the component.

What do people think of this? I really do not think we need to seek another third party package to resolve this.

**Comment by ghost** [comment]
@12 Adding a `white` background works fine. I guess in my suggestions I was trying to preserve the loading behaviour we currently have . We might want to check with **UX** wether its ok to have a white background when the image is loading

**Comment by 12** [comment]
@sadickisaac You don't see a white background whilst the image is loading, you still see the BBC blocks.
